src/main/golem/util/validation/bounds.kt

```diff
@@ -0,0 +1,49 @@
+@file:JvmName("BoundsValidation")
+
+package golem.util.validation
+import golem.validateMatrices
+import golem.*
+
+/**
+ * Maximum value for individual elements in the matrix.
+ */
+var ValidationContext.max : Double
+    get() { throw UnsupportedOperationException() }
+    set(value) { max(value) }
+
+/**
+ * Ensure all of the elements in the current matrix are >= the given value.
+ * @param value The maximum value required of all elements in the matrix.
+ */
+fun ValidationContext.max(value: Double) : ValidationContext {
+    if (!validateMatrices) return this
```

Comment:
The returns are there for the "chainable function" alternative syntax. `.dim(3, 4).min(4).max(5)`.